ui/src/reducers/user.js

```diff
@@ -1,15 +1,19 @@
 import {
   USER_REQUEST,
   USER_SUCCESS,
-  USER_FAILURE
+  USER_FAILURE,
+  USER_AUTHENTICATE_SUCCESS,
+  USER_AUTHENTICATE_FAILURE
 } from '../actions/types.js';
 
 const initialState = {
   user: {
     username: '',
     email: ''
   },
-  isUserRequestFailed: false
+  isUserRequestFailed: false,
+  isLoggedIn: false,
+  isLoading: true
```

Comment:
What is the isLoading value used for?

Reply:
See [this line](https://github.com/aerogear/mobile-security-service/pull/154/files#diff-b783c7f57868724b8b576eb3f530dfb1R27).

Without a separate `isLoading` state, all we would have to work with is `isLoggedIn`. `isLoggedIn` is a boolean which starts off as false. As it starts off as false, the page would redirect straight away because it would see `isLoggedIn === false` and do the redirect. 

`isLoading` prevents this. Only when the application knows that a) we have checked if the user is logged in (setting `isLoading` to false) and b) the user is not logged in (setting `isLoggedIn` to false) will a redirect happen.